src/test/java/org/prebid/server/bidder/pangle/PangleBidderTest.java

```diff
@@ -470,6 +479,102 @@ public void makeBidsShouldReturnErrorsAndResult() throws JsonProcessingException
         assertThat(result.getErrors()).hasSize(1);
     }
 
+    @Test
+    public void makeHttpRequestsShouldUpdateImpExtWithNetworkIds() {
+        // given
+        final BidRequest bidRequest = BidRequest.builder()
+                .imp(singletonList(Imp.builder()
+                        .banner(Banner.builder().w(1).h(1).build())
+                        .ext(mapper.valueToTree(WrappedImpExtBidder.of(null, ExtImpPangle.of(
+                                "token", "2", "2"),
+                                null, true, NetworkIds.of("1", "1"))))
+                        .build()))
+                .build();
+
+        // when
+        final Result<List<HttpRequest<BidRequest>>> result = pangleBidder.makeHttpRequests(bidRequest);
+
+        // then
+        final ObjectNode expectedExt = mapper
+                .valueToTree(WrappedImpExtBidder.of(null, ExtImpPangle.of(
+                        "token", "2", "2"),
+                        1, true, NetworkIds.of("2", "2")));
+        assertThat(result.getErrors()).isEmpty();
+        assertThat(result.getValue())
+                .extracting(HttpRequest::getPayload)
+                .flatExtracting(BidRequest::getImp)
+                .extracting(Imp::getExt)
+                .containsExactly(expectedExt);
+    }
+
+    @Test
+    public void makeHttpRequestsShouldThrowExceptionIfAppIdIsNotPresentAndPlacementIdIsPresent() {
```

Comment:
makeHttpRequestsShouldAddErrorAndSkipImpressionIfAppIdIsNotPresentAndPlacementIdIsPresent

Also pls check for same occurrences